### ISSUE TITLE
Runtime sitekey configuration

### ISSUE THREAD

**Issue by niksoft123** [chvarkov/google-recaptcha#139]
Hello,

I can't find a way of dynamic/runtime setup of the sitekey of the recaptcha without restarting the whole application.
Imagine we have a separate application which behave like Admin and there is possibility to change the sitekey too. 
Now the lib works fine, when via this Admin I'm switching it ON/OFF.
I don't know if it is possible to have something like skipIf callback, but for sitekey ?

Best Regards

**Comment by chvarkov** [comment]
Hi @niksoft123, thank you for the issue it's interesting point.
As solution I see some injectable class that will represent configuration reference and have a methods to update recaptcha keys, `skipIf` and `score`. It should solve your case and save backward compatibility.

For example `RecaptchaConfigRef`:
```
class RecaptchaConfigRef {
	get valueOf(): GoogleRecaptchaModuleOptions;

	setSecretKey(secretKey: string): this;

	setEnterpriseOptions(options: GoogleRecaptchaEnterpriseOptions): this;

	setScore(score: ScoreValidator): this;

	setSkipIf(skipIf: SkipIfValue): this;
}
```

In this way it can be used in your app:
```
@Injectable()
export class AdminService implements OnApplicationBootstrap {
	constructor(private readonly recaptchaConfigRef: RecaptchaConfigRef) {
	}

	async onApplicationBootstrap(): Promise<void> {
		// TODO: Pull recaptcha configs from your database

		this.recaptchaConfigRef
			.setSecretKey('SECRET_KEY_VALUE')
			.setScore(0.3);
	}

	async updateSecretKey(secretKey: string): Promise<void> {
		// TODO: Some your logic

		this.recaptchaConfigRef.setSecretKey(secretKey);
	}
}
```

After call `this.recaptchaConfigRef.setSecretKey(...)` `@Recaptcha` guard will use new secret key.
What do you think about this? Is it will solve your issue?

**Comment by niksoft123** [comment]
Hello, 

and thanks for time spent under this "issue".
Yes, absolutely solved my case.
If you think that it will be helpful for such a library setup it will be fine, but if not, it will be ok for me and for people like me just to update the spec/documentation with some similar example.

Thanks a lot for your time !

Best Regards

**Comment by chvarkov** [comment]
Hey @niksoft123 new version has been released just now. `RecaptchaConfigRef` will be available since `v3.7.0`.

**Comment by niksoft123** [comment]
Thank you very much !
